### PR TITLE
[cifuzz] Only grab github-specific env vars on github

### PR DESCRIPTION
**Pull request by jonathanmetzman**
This removes need for non-OSS-Fuzz users to fake using OSS-Fuzz.

Fixes: #6129

### CODE REVIEW

File: infra/cifuzz/config_utils.py [hunk 1]
```diff
@@ -88,7 +59,62 @@ def _get_language():
 # pylint: disable=too-few-public-methods,too-many-instance-attributes
 
 
-class ConfigurationError(Exception):
+class CiEnvGetter:
```
Comment (oliverchang) [inline infra/cifuzz/config_utils.py:62]: Let's structure it like this:

```python

class CiEnvironment:
   WORKSPACE_VAR = 'WORKSPACE'
   REPOSITORY_VAR  = 'REPOSITORY'
   TOKEN_VAR = 'TOKEN'
   GIT_SHA_VAR = 'GIT_SHA'

  def workspace(self):
    return os.getenv(self.WORKSPACE_VAR)

  def owner_and_repo(self):
   repository = os.getenv(self.REPOSITORY_VAR)
   ...

  def project_src_path(self):
    return os.getenv('PROJECT_SRC_PATH')


class GitHubEnvironment(CiEnvironment):
   WORKSPACE_VAR = 'GITHUB_WORKSPACE'
   REPOSITORY_VAR  = 'GITHUB_REPOSITORY'
   TOKEN_VAR = 'GITHUB_TOKEN'
   GIT_SHA_VAR = 'GITHUB_SHA'

   def project_src_path(self):
      # On GitHub, they don't know the absolute path, it is relative to
      # |workspace|.
     return os.path.join(self.workspace(), super().project_src_path())
```

This way, there's a clearer separation and it becomes much easier to extend support for other CI systems in the future if necessary.

File: infra/cifuzz/config_utils.py [hunk 2]
```diff
@@ -88,7 +59,62 @@ def _get_language():
...
+    # On GitHub this includes owner and repo name.
+    repository = os.getenv(variable, '')
+    # Use os.path.split. When GITHUB_REPOSITORY just contains the name of the
+    # repo, this will return a tuple containing an empty string and the repo
+    # name. When GITHUB_REPOSITORY contains the repo owner followed by a slash
+    # and then the repo name, it will return a tuple containing the owner and
+    # repo name.
+    return os.path.split(repository)
+
+  def get_project_src_path(self, workspace):
```
Comment (oliverchang) [inline infra/cifuzz/config_utils.py:100]: no need to require a `workspace` variable here, as it's leaking an implementation detail for github env handling. We can just read the workspace again in GitHub's implementation (see above comment).

**Review by oliverchang — COMMENTED** [review]
Thanks! have a suggestion to make this implementation cleaner and more easily extensible.

File: infra/cifuzz/config_utils.py (hunk 1 above)
Comment (jonathanmetzman) [reply infra/cifuzz/config_utils.py]: Done. PTAL.

File: infra/cifuzz/config_utils.py (hunk 2 above)
Comment (jonathanmetzman) [reply infra/cifuzz/config_utils.py]: Done.

**Review by oliverchang — APPROVED** [review]

**Pull request merged**